Cannot directly create `?int(none)` Option

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 5ec9bb5.9e71e32
OS: linux, Linux version 6.6.8-arch1-1 (linux@archlinux) (gcc (GCC) 13.2.1 20230801, GNU ld (GNU Binutils) 2.41.0) #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000
Processor: 16 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i5-1240P

getwd: /home/ge
vexe: /home/ge/.vlang/v
vexe mtime: 2024-12-14 17:38:37

vroot: OK, value: /home/ge/.vlang
VMODULES: OK, value: /home/ge/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.47.1
Git vroot status: weekly.2024.50-36-g9e71e324
.git/config present: true

CC version: cc (GCC) 14.2.1 20240910
emcc version: N/A
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg create_int_option.vv`

```v
fn main() {
	n := ?int(none)
	println(n)
}
```

**What did you expect to see?**

`Option(none)` in output. This is described in [docs](https://docs.vlang.io/type-declarations.html#optionresult-types-and-error-handling).

Aliased `int` works:

```v
type Int = int

fn main() {
	n := ?Int(none)
	println(n)
}
// Option(none)
```

**What did you see instead?**

```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/create_int_option.01JF34102EGSCVY1PDBCB8Z2D7.tmp.c:1267: warning: implicit declaration of function 'str_intp'
cc: /tmp/v_1000/create_int_option.01JF34102EGSCVY1PDBCB8Z2D7.tmp.c:1267: error: cannot convert 'int' to 'struct string'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzVkYzNlYzFlMDYzMmRkMDhlMzllMzkiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.KezeKBMhHHPkAjbkh2Y4uhBfUTZcq8tfWopFx4mz0cA">Huly&reg;: <b>V_0.6-21600</b></a></sub>

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1422,6 +1422,9 @@ fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) ast.
 			if !c.table.sym(c.unwrap_generic(node.args[0].typ)).has_method('str') {
 				c.table.used_features.auto_str = true
 			} else {
+				if node.args[0].typ.has_flag(.option) {
+					c.table.used_features.option_or_result = true
+				}
 				c.table.used_features.print_types[node.args[0].typ.idx()] = true
 			}
 			if node.args[0].typ.is_ptr() {

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1422,7 +1422,7 @@ fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) ast.
 			if !c.table.sym(c.unwrap_generic(node.args[0].typ)).has_method('str') {
 				c.table.used_features.auto_str = true
 			} else {
-				if node.args[0].typ.has_flag(.option) {
+				if node.args[0].typ.has_option_or_result() {
 					c.table.used_features.option_or_result = true
 				}
 				c.table.used_features.print_types[node.args[0].typ.idx()] = true

--- a/vlib/v/tests/skip_unused/option_int_println.run.out
+++ b/vlib/v/tests/skip_unused/option_int_println.run.out
@@ -1,0 +1,1 @@
+Option(none)

--- a/vlib/v/tests/skip_unused/option_int_println.skip_unused.run.out
+++ b/vlib/v/tests/skip_unused/option_int_println.skip_unused.run.out
@@ -1,0 +1,1 @@
+Option(none)

--- a/vlib/v/tests/skip_unused/option_int_println.vv
+++ b/vlib/v/tests/skip_unused/option_int_println.vv
@@ -1,0 +1,4 @@
+fn main() {
+	n := ?int(none)
+	println(n)
+}

